File: W/1637-widest-vertical-area-between-two-points-containing-no-points/main.go
package main

import (
	"fmt"
	"sort"
)

func maxWidthOfVerticalArea(points [][]int) int {
	xm := map[int]int{}
	for _, v := range points {
		xm[v[0]]++
	}
	var cord []int
	for k := range xm {
		cord = append(cord, k)
	}
	if len(cord) <= 1 {
		return 0
	}
	sort.Ints(cord)

	max := -1
	for i := 1; i < len(cord); i++ {
		if cord[i]-cord[i-1] > max {
			max = cord[i] - cord[i-1]
		}
	}
	return max
}

func main() {
	fmt.Println()
}

File: W/1637-widest-vertical-area-between-two-points-containing-no-points/main.rs
struct Solution {}

impl Solution {
  pub fn max_width_of_vertical_area(points: Vec<Vec<i32>>) -> i32 {
    let mut x: Vec<i32> = points.iter().map(|v| v[0]).collect::<Vec<i32>>();
    x.sort();
    *(0..x.len())
      .map(|i| match i {
        0 => 0,
        _ => x[i] - x[i - 1],
      })
      .collect::<Vec<i32>>()
      .iter()
      .max()
      .unwrap() as i32
  }
}
